substrate/client/rpc-spec-v2/src/chain_head/subscription/mod.rs

```diff
@@ -43,30 +64,55 @@ impl<Block: BlockT, BE: Backend<Block>> SubscriptionManagement<Block, BE> {
 		global_max_pinned_blocks: usize,
 		local_max_pin_duration: Duration,
 		max_ongoing_operations: usize,
+		max_follow_subscriptions_per_connection: usize,
 		backend: Arc<BE>,
 	) -> Self {
 		SubscriptionManagement {
-			inner: RwLock::new(SubscriptionsInner::new(
+			inner: Arc::new(RwLock::new(SubscriptionsInner::new(
 				global_max_pinned_blocks,
 				local_max_pin_duration,
 				max_ongoing_operations,
 				backend,
-			)),
+			))),
+			rpc_connections: RpcConnections::new(max_follow_subscriptions_per_connection),
 		}
 	}
 
-	/// Insert a new subscription ID.
+	/// Create a new instance from the inner state.
 	///
-	/// If the subscription was not previously inserted, returns the receiver that is
-	/// triggered upon the "Stop" event. Otherwise, if the subscription ID was already
-	/// inserted returns none.
-	pub fn insert_subscription(
+	/// # Note
+	///
+	/// Used for testing.
+	#[doc(hidden)]
```

Comment:
If only used for testing, you could have `#[cfg(test)]` here instead to guarantee that (and make it super obvious) :)